### PR TITLE
Fix duplicate OOI references in result in origin

### LINKED ISSUE #3508: Duplicate OOI references in result in origin
**Describe the bug**
Duplicate references in result in origin.

Example (generated by main b098d8d2ee2f7bdbb4c7d98e7125ae3c16f43bd7):
```
{
  "method": "dns-alias-resolving",
  "origin_type": "inference",
  "result": [
    "ResolvedHostname|internet|engine.entree.kennisnet.nl|internet|145.97.35.101",
    "ResolvedHostname|internet|engine.entree.kennisnet.nl|internet|145.97.35.101",
    "ResolvedHostname|internet|engine.entree.kennisnet.nl|internet|2001:610:2d8:300::35:101",
    "ResolvedHostname|internet|engine.entree.kennisnet.nl|internet|2001:610:2d8:300::35:101"
  ],
  "source": "Hostname|internet|ocauthenticatie.entree.kennisnet.nl",
  "source_method": null,
  "task_id": null,
  "type": "Origin",
  "xt/id": "Origin|inference|dns-alias-resolving|Hostname|internet|ocauthenticatie.entree.kennisnet.nl"
}
```
Note the duplication's in the results.

As we are not expecting this, we might get undefined behavior. Probably `result` should be a `set` instead of a `list`.
https://github.com/minvws/nl-kat-coordination/blob/ba809d0ce61c3a0f28940c97bd048582c657481a/octopoes/octopoes/models/origin.py#L16-L22

**To Reproduce**

1. Enable [DNSRecords(A, AAAA), NMap TCP, SSLCertificates, WebpageAnalysis] boefjes
2. Scan URL("http://metaplus.kennisnet.nl/") on L2
3. Wait
4. Observe Origin exampled above with `octopoes/tools/xtdb-cli.py --node 2 query '{:query {:find [(pull ?var [*])] :where [[?var :type "Origin"]]}}' | jq -C  | less -R`

**Expected behavior**
```
{
  "method": "dns-alias-resolving",
  "origin_type": "inference",
  "result": [
    "ResolvedHostname|internet|engine.entree.kennisnet.nl|internet|145.97.35.101",
    "ResolvedHostname|internet|engine.entree.kennisnet.nl|internet|2001:610:2d8:300::35:101"
  ],
  "source": "Hostname|internet|ocauthenticatie.entree.kennisnet.nl",
  "source_method": null,
  "task_id": null,
  "type": "Origin",
  "xt/id": "Origin|inference|dns-alias-resolving|Hostname|internet|ocauthenticatie.entree.kennisnet.nl"
}
```

**OpenKAT version**
main b098d8d2ee2f7bdbb4c7d98e7125ae3c16f43bd7

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes

Fix duplicate OOI references in result in origin

### Issue link

Closes #3508

### Demo

N/A

### QA notes

Two methods:
* Use VisualOctopoesStudio on a typical run to check there are no duplicate references in results
* Use `octopoes/tools/xtdb-cli.py query '{:query {:find [(pull ?var [*])] :where [[?var :type "Origin"]]}}' | jq -C  | less -R` and check there are no duplicates in a typical run

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/octopoes/repositories/origin_repository.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def serialize(cls, origin: Origin) -> dict[str, Any]:
         data = origin.model_dump()
         data[cls.pk_prefix] = origin.id
         data["type"] = origin.__class__.__name__
+        data["result"] = list(dict(zip(data["result"], [None] * len(data["result"]))).keys())
```
Comment (ammar92) [inline octopoes/octopoes/repositories/origin_repository.py:66]: It took me a few times reading this and the PR title to understand it; but if I'm correct you're trying to remove duplicates in the `data["result"]` list, while also keeping the order?

If that's the case, I'd like to suggest this snippet as it's a bit shorter and also a common approach:
```suggestion
        data["result"] = list(dict.fromkeys(data["result"]))
```

File: octopoes/tests/integration/test_io.py
Comment (ammar92) [inline octopoes/tests/integration/test_io.py:1]: Clear 👍

**Review by ammar92 — COMMENTED** [review]
Nice work! Left one tiny suggestion

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Seems to work. Not seeing any duplicate references in the tried hosts.

### What doesn't work:
N/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**